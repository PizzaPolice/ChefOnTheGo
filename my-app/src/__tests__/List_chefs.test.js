import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import ListChefs from '../components/list_chefs.js'


describe('Login Component', () => {
    it('Login should render without throwing an error', () => {
        expect(shallow(<ListChefs />).exists(<form className='Login'></form>)).toBe(true)
      })
      it("should render initial layout", () => {
        const component = shallow(<ListChefs />);
        expect(component.getElements()).toMatchSnapshot();
        })
    });



    describe('Submit Button', () => {
        it('should be defined', () => {
          expect(ListChefs).toBeDefined();
        });

        it('should render correctly', () => {
          const button = shallow(
            <ListChefs name='btn back-btn' />
          );
          expect(button).toMatchSnapshot();
        });
    });
