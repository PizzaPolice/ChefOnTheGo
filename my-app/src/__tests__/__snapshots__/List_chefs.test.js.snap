// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListChefs Component should render initial layout 1`] = `
Array [
  <div
    className="outer-panel"
>
    <div
        className="panel-body"
    >
        <h2
            className="title"
        >
            List of Chefs
        </h2>
        <div
            className="input-fields"
        >
            <ChefList
                chefs={
                    Array [
                        Object {
                          "bio": "Chef for 5 years",
                          "name": "Jae",
                          "rating": "5 stars",
                        },
                        Object {
                          "bio": "Chef for 6 years",
                          "name": "Kevin",
                          "rating": "5 stars",
                        },
                        Object {
                          "bio": "Chef for 7 years",
                          "name": "Audrey",
                          "rating": "5 stars",
                        },
                      ]
                }
            />
        </div>
    </div>
    <div
        className="panel-footer"
    >
        <Link
            replace={false}
            to="login"
        >
            <button
                className="btn back-btn"
            >
                Go Back
            </button>
        </Link>
        <Link
            replace={false}
            to="chat"
        >
            <button
                className="btn next-btn"
            >
                Chat!
            </button>
        </Link>
    </div>
</div>,
]
`;
